localstorage/firebase-worker.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Dedicated web worker that loads Firebase off of the main browser thread.
+ * Each tab will run an instance of Firestore via the web worker. The worker
+ * receives messages from your main page and performs Firestore operations.
+ * 
+ * Must be loaded and initialized from your main page:
+ * dedicatedWorker = new Worker("firebase-worker.js");
+ */
+
+importScripts('../firebase/firebase-app.js');
+importScripts('../firebase/firebase-firestore.js');
+
+var firebaseConfig = {
+  apiKey: "api-key",
+  authDomain: "project-id.firebaseapp.com",
+  databaseURL: "https://project-id.firebaseio.com",
+  projectId: "project-id",
+  storageBucket: "project-id.appspot.com",
+  messagingSenderId: "sender-id",
+  appId: "app-id",
+  measurementId: "G-measurement-id",
+};
+
+/** The Firestore instance. */
+self.db = null;
+self.onmessage = async function(e) {
+  // Do Firebase work here.
```

Comment:
This could be interpreted as if you can/should do Firebase work here (directly inside onmessage), even though we haven't necessarily run initializeApp() yet, etc.  Maybe move this down and rephrase (see below comment).